src/libMessage/Messenger.cpp

```diff
@@ -113,6 +117,162 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
   Serializable::SetNumber<T>(dst, offset, number, S);
 }
 
+// ============================================================================
+// Functions to check for fields in primitives that are used for persistent
+// storage. Remove fields from the checks once they are deprecated.
+// ============================================================================
+
+inline bool CheckFieldsInProtoBlockLink(const ProtoBlockLink& protoBlockLink) {
```

Comment:
Maybe we can change the API name to CheckRequiredFieldsxxx(), because this function is to replace the "required" field.